architecture/ambient/lifecycle.md

```diff
@@ -0,0 +1,136 @@
+# Ztunnel Lifecyle
+
+This document describes the lifecycle of various aspects of Ztunnel, and its relationship to workloads and the CNI.
+
+At a high level, our goal is to provide complete connectivity to a workload throughout its entire lifetime.
+Failing to do so can be an availability risk (if we deny traffic that should succeed) or a security risk (if we allow traffic that should be denied).
+
+## High level overview
+
+At a high level, the relevant components look as such:
+
+```mermaid
+flowchart TD
+    CNIP["CNI Plugin"]
+    CNI["CNI Agent"]
+    Z["Ztunnel"]
+    Pods["Pods"]
+    CNI --Programs--> Z
+    CNI --Watches--> Pods
+    Z --Runs Within--> Pods
+    CNIP --Calls--> CNI
+```
+
+The CNI Plugin is a binary installed as a [CNI plugin](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/) on the node.
+Kubernetes is responsible for invoking this when a Pod is being started (before the containers run).
+When this occurs, the plugin will call out to the CNI Agent to program the network.
+This includes setting up networking rules both within the pod network namespace, and also on the host network.
+
+An alternative flow is when a pod is enabled after startup.
```

Comment:
Enabled after which startup ? How can that happen ( the CNI binary will be always called).
You mean 'if the CNI binary is installed later ' ?

Reply:
Its if the namespace/pod is enrolled after it starts. By label, etc